src/Basque/Mews.Fiscalizations.Basque/DtoToModelConverter.cs

```diff
@@ -41,4 +76,41 @@ private static T ParseEnum<T>(string value)
             _ => throw new NotImplementedException($"{value} is not implemented in {typeof(T).Name}.")
         );
     }
+    
+    private static InvoiceState ConvertBizkaiaState(EstadoRegistroEnum state)
+    {
+        return state.Match(
+            EstadoRegistroEnum.Correcto, _ => InvoiceState.Received,
+            EstadoRegistroEnum.AceptadoConErrores, _ => InvoiceState.Received,
+            EstadoRegistroEnum.Incorrecto, _ => InvoiceState.Refused,
+            EstadoRegistroEnum.Anulado, _ => InvoiceState.Refused
+        );
+    }
+
+    private static ErrorCode ConvertBizkaiaErrorCodes(string bizkaiaErrorCode)
+    {
+        switch (bizkaiaErrorCode)
+        {
+            case "B4_2000001":
+            case "B4_1000001": return ErrorCode.XsdSchemeViolation;
+            case "B4_2000003":
+            case "B4_1000005": return ErrorCode.DuplicateInvoice;
+            case "B4_2000000": return ErrorCode.MissingMandatoryData;
+            case "B4_1000004": return ErrorCode.ServerErrorTryAgain;
+            case "B4_2000013": return ErrorCode.TaxIdentifierCountryCodeDoesntMatchCountryCodeField;
+            case "B4_2000011": return ErrorCode.InvalidReceiverTaxIdentifierFormat;
+            case "B4_2000012": return ErrorCode.MissingCountryCode;
+            case "B4_2000008": return ErrorCode.InvoiceIssueDateGreaterThanCurrentDate;
+            case "B4_2000016": return ErrorCode.CorrectedInvoiceMustNotBeReportedWhenCorrectiveInvoiceIsNotReported;
+            case "B4_2000045": return ErrorCode.CorrectedInvoiceNotIndicated;
+            case "B4_2000070": return ErrorCode.InvalidSignatureOrSigningCertificate;
+            case "B4_2000060": 
+            case "B4_2000061": return ErrorCode.InvalidOrMissingInvoiceChain;
+            case "B4_2000038": return ErrorCode.BreakdownMustHaveProvisionOrDeliveryOrBoth;
+            case "B4_2000030": return ErrorCode.InvoiceMustContainAtLeastOneExemptOrNonExemptParty;
+
+            default: throw new NotImplementedException("We currently have no support for this error code.");
```

Comment:
This would look much much better with a switch expression (pattern matching)

or feel free to use FuncSharp match